src/jdom/bus.ts

```diff
@@ -731,23 +770,6 @@ export class JDBus extends JDNode {
     withTimeout<T>(timeout: number, p: Promise<T>): Promise<T> {
         return new Promise<T>((resolve, reject) => {
             let done = false
-            let tid = setTimeout(() => {
-                if (!done) {
-                    done = true
-                    if (!this.connected) {
-                        // the bus got disconnected so all operation will
-                        // time out going further
-                        this.emit(TIMEOUT_DISCONNECT)
-                        resolve(undefined);
-                    }
-                    else {
-                        // the command timed out
-                        this.emit(TIMEOUT)
-                        this.emit(ERROR, "Timeout (" + timeout + "ms)");
-                        resolve(undefined);
-                    }
-                }
-            }, timeout)
```

Comment:
you removed the timeout from withTimeout?

Reply:
oh TypeScript is too smart and removed it because it things tid is not used.

Reply:
I guess we should use it in clearTimeout(), but anyway removing a call with side effects doesn't sound like a good idea